packages/rich-text/src/plugins/Editor/index.ts

```diff
@@ -0,0 +1,33 @@
+import { Editor, Element, Text } from 'slate';
+import { CustomEditor, CustomElement } from '../../types';
+
+export function withEditorPlugin(editor: CustomEditor): CustomEditor {
+  const { isVoid: originalIsVoid } = editor;
+
+  function isBlockSelected(type: string): boolean {
+    const [match] = Array.from(
+      Editor.nodes(editor, {
+        match: (node) => Element.isElement(node) && (node as CustomElement).type === type,
+      })
+    );
+    return !!match;
+  }
+
+  function isVoid(element): boolean {
+    return element.isVoid || originalIsVoid(element);
```

Comment:
`originalIsVoid` comes ultimately from the schema, right? When would `element.isVoid` be false but not the schema `isVoid`?